pkg/controller/activegate/activegate_update.go

```diff
@@ -14,15 +15,33 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
-func (r *ReconcileActiveGate) updatePods(
+type updateService interface {
+	FindOutdatedPods(
+		r *ReconcileActiveGate,
+		logger logr.Logger,
+		instance *dynatracev1alpha1.ActiveGate) ([]corev1.Pod, error)
+	IsLatest(logger logr.Logger, image string, imageID string, imagePullSecret *corev1.Secret) (bool, error)
+	UpdatePods(
+		r *ReconcileActiveGate,
+		pod *corev1.Pod,
+		instance *dynatracev1alpha1.ActiveGate,
+		secret *corev1.Secret) (*reconcile.Result, error)
+}
+
+type activeGateUpdateService struct{}
```

Comment:
I don't mind an extra object for a particular purpose, but two objects querying each other, e.g., the reconciler calling the service, and then service calling the reconciler is confusing.

I'd suggest either leaving them on the same type, or focusing the service to avoid contacting the reconciler.